test/cql-pytest/test_describe.py

```diff
@@ -7,12 +7,572 @@
+    fn_schema = "(val1 int, val2 int)"
+    fn_null = "RETURNS NULL ON NULL INPUT"
+    fn_return = "RETURNS int"
+    fn_language = "LANGUAGE lua"
+    fn_body = "return val1 + val2"
+
+    with new_function(cql, test_keyspace, f"{fn_schema} {fn_null} {fn_return} {fn_language} AS '{fn_body}'") as fn:
+        agg_body = f"(int) SFUNC {fn} STYPE int INITCOND 0"
+        with new_aggregate(cql, test_keyspace, agg_body):
+            assert_element_listing(cql, "aggregates", "aggregate_name", lambda r: r.aggregate_name)
+
+# Test that `DESC ONLY KEYSPACE {ks}` contains appropriate create statement for keyspace
+# This test compares the content of `system_schema.keyspaces` table.
+def test_desc_keyspace(cql, random_seed):
+    with new_random_keyspace(cql) as ks:
+        desc = cql.execute(f"DESC ONLY KEYSPACE {ks}")
+        desc_stmt = desc.one().create_statement
+
+        new_ks = unique_name()
+        new_desc_stmt = desc_stmt.replace(ks, new_ks)
+
+        try:
+            cql.execute(new_desc_stmt)
+            ks_row = cql.execute(f"""
+                SELECT durable_writes, replication
+                FROM system_schema.keyspaces
+                WHERE keyspace_name='{ks}'
+            """).one()
+            new_ks_row = cql.execute(f"""
+                SELECT durable_writes, replication
+                FROM system_schema.keyspaces
+                WHERE keyspace_name='{new_ks}'
+            """).one()
+            assert ks_row == new_ks_row
+        finally:
+            cql.execute(f"DROP KEYSPACE {new_ks}")
+# Test that `DESC ONLY KEYSPACE {ks}` contains appropriate create statement for keyspace
+# This test compares the content of `system_schema.scylla_keyspaces` table, thus the test
+# is `scylla_only`.
+def test_desc_scylla_keyspace(scylla_only, cql, random_seed):
+    with new_random_keyspace(cql) as ks:
+        desc = cql.execute(f"DESC ONLY KEYSPACE {ks}")
+        desc_stmt = desc.one().create_statement
+
+        new_ks = unique_name()
+        new_desc_stmt = desc_stmt.replace(ks, new_ks)
+
+        try:
+            cql.execute(new_desc_stmt)
+            ks_scylla_row = cql.execute(f"""
+                SELECT storage_options, storage_type
+                FROM system_schema.scylla_keyspaces
+                WHERE keyspace_name='{ks}'
+            """).one()
+            new_ks_scylla_row = cql.execute(f"""
+                SELECT storage_options, storage_type
+                FROM system_schema.scylla_keyspaces
+                WHERE keyspace_name='{new_ks}'
+            """).one()
+            assert ks_scylla_row == new_ks_scylla_row
+        finally:
+            cql.execute(f"DROP KEYSPACE {new_ks}")
+
+# Test that `DESC TABLE {tbl}` contains appropriate create statement for table
+# This test compares the content of `system_schema.tables` and `system_schema.columns` tables.
+def test_desc_table(cql, test_keyspace, random_seed):
+    with new_random_table(cql, test_keyspace) as tbl:
```

Comment:
I think this "random table" misses one important case. 
Imagine that some string inside the tables definition (e.g., the "comment", but there are many other strings inside various maps etc.) contains a single-quote or newline or some other special character. We need to make sure that "DESC TABLE" quotes this string in a way it will be usable as a CQL command. We don't want to have our own version of SQL injection attacks if someone uses `; DROP TABLES` as a comment on a table and then the DESCRIBE TABLE is used to save a backup of this table - and then restoring the backup will cause the table to be deleted.

I would write this test as a stand-alone test. I think the idea of the "random table" can only be pushed so far. If we suspect a specific type of bug we should have a special test specifically for it, and don't need to insert it into some big random-generation code that nobody will later understand what it actually covers and what it doesn't.

Reply:
You forgot to add a link to https://xkcd.com/327

Reply:
I checked that and wrote a test for it. The test creates and describes a table which contains column with type requiring quoting and also mentioned comment